### PR TITLE
Testing Framework

### PR DESCRIPTION
**Pull request by bbrockbernd**
Added a testing framework. Tests can be defined in src/test/resources/data.json

### CODE REVIEW

File: src/main/java/org/jetbrains/research/kotlinrminer/api/GitService.java [hunk 1]
```diff
@@ -66,6 +67,59 @@ public void fileTreeDiff(Repository repository, RevCommit currentCommit, List<St
...
+        Repository repository;
+        if (folder.exists()) {
+            RepositoryBuilder builder = new RepositoryBuilder();
+            repository = builder
+                .setGitDir(new File(folder, ".git"))
+                .readEnvironment()
+                .findGitDir()
+                .build();
+
+            //logger.info("Project {} is already cloned, current branch is {}", cloneUrl, repository.getBranch());
```
Comment (onewhl) [inline src/main/java/org/jetbrains/research/kotlinrminer/api/GitService.java]: Please, either add a logger or delete the comment.

File: src/main/java/org/jetbrains/research/kotlinrminer/api/GitService.java [hunk 2]
```diff
@@ -66,6 +67,59 @@ public void fileTreeDiff(Repository repository, RevCommit currentCommit, List<St
...
+            Git git = Git.cloneRepository()
+                .setDirectory(folder)
+                .setURI(cloneUrl)
+                .setCloneAllBranches(true)
+                .call();
+            repository = git.getRepository();
+            //logger.info("Done cloning {}, current branch is {}", cloneUrl, repository.getBranch());
+        }
+
+//		if (branch != null && !repository.getBranch().equals(branch)) {
```
Comment (onewhl) [inline src/main/java/org/jetbrains/research/kotlinrminer/api/GitService.java]: Please, delete unused code

**Comment by onewhl** [comment]
@bbrockbernd, good job, thank you! Please, take a look at the comments I left.

File: src/test/resources/data.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  "repository" : "https://github.com/JetBrains-Research/kotlin-refactoring-samples.git",
+  "sha1" : "38448785791cffce8c453b04bcc3e217bb41f1e8",
+  "refactorings" : [{
+    "type" : "Rename Class"
+  }]
+}, {
+  "repository" : "https://github.com/JetBrains-Research/kotlin-refactoring-samples.git",
+  "sha1" : "38448785791cffce8c453b04bcc3e217bb41f1e8",
+  "refactorings" : [{
+    "type" : "This test should fail!"
```
Comment (onewhl) [inline src/test/resources/data.json]: Is there a reason for this test?

**Review by onewhl — APPROVED** [review]

File: src/test/resources/data.json (hunk 3 above)
Comment (bbrockbernd) [reply src/test/resources/data.json]: Yeah this was to test the tests

**Pull request merged**